contracts/GSN/Context.sol

```diff
@@ -14,13 +14,12 @@ contract Context {
     // Empty internal constructor, to prevent people from mistakenly deploying
     // an instance of this contract, which should be used via inheritance.
     constructor () internal { }
-    // solhint-disable-previous-line no-empty-blocks
 
-    function _msgSender() internal view returns (address payable) {
+    function _msgSender() internal view virtual returns (address payable) {
```

Comment:
Yes, a function being `virtual` is a giveaway that it should always be used instead of the underlying data. 

Note however that we generally don't have `internal view` functions: the only exception I've found is `ERC721._exists(tokenId`).